### ISSUE TITLE
Which loss I can choose when there are multiple positive and negative samples for a query

### ISSUE THREAD

**Issue by Doragd** [KevinMusgrave/pytorch-metric-learning#492]
Hi, Kevin.
Currently, I am studying a text classification task. 
I want to use one of the metric losses with the classification loss.
There are multiple positive and negative samples for a query.
It seems that SupConLoss can solve it. Is it correct?

By the way, I don't know if I should normalize the features before using the metric loss, like that
```python
out = model(x) # get representation of texts

#should I normalize the out?
metric_loss = loss(l2_normalize(out))

logits = classifier(out) # convert to logits
classification_loss = ce_loss(logits, labels)
```

**I also don't know the pos_mask and neg_mask in SupConLoss.** 

Could you help me to solve it?

**Comment by KevinMusgrave** [comment]
Yes you can use SupConLoss:

```python
metric_loss = loss(out, labels)
```

You don't need to normalize the features. That is done inside the loss function by default.

**Comment by Doragd** [comment]
Thanks.
